pkgs/servers/pleroma-otp/default.nix

```diff
@@ -0,0 +1,68 @@
+{ stdenv
+, autoPatchelfHook
+, makeWrapper
+, ncurses
+, openssl
+, zlib
+, fetchurl
+, unzip
+, nixosTests
+}:
+stdenv.mkDerivation {
+  pname = "pleroma-otp";
+  version = "2.1.2";
+
+  # To find the latest binary release stable link, have a look at
+  # the CI pipeline for the latest commit of the stable branch
+  # https://git.pleroma.social/pleroma/pleroma/-/tree/stable
+  src = {
+    aarch64-linux = fetchurl {
+      url = "https://git.pleroma.social/pleroma/pleroma/-/jobs/154866/artifacts/download";
+      sha256 = "1mwbhbfi01nyq8d96yd3f2qfizj3zn8rx73967vqnlxh7dshc6ya";
+    };
+    x86_64-linux = fetchurl {
+      url = "https://git.pleroma.social/pleroma/pleroma/-/jobs/154862/artifacts/download";
+      sha256 = "16352ail4w1v0im5w1aq0z4lb1k85qa2qrr44pq12h0l6m05bc0z";
+    };
+  }."${stdenv.hostPlatform.system}";
+
+  nativeBuildInputs = [ unzip ];
+
+  buildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    ncurses
+    openssl
+    zlib
+  ];
+
+  # mkDerivation fails to detect the zip nature of $src due to the
+  # missing .zip extension.
+  # Let's unpack the archive explicitely.
+  unpackCmd = "unzip $curSrc";
+
+  installPhase = ''
+    mkdir $out
+    cp -r * $out'';
+
+  # Pleroma is using the project's root path (here the store path)
+  # as its TMPDIR.
+  # Patching it to move the tmp dir to the actual tmpdir
+  postFixup = ''
+    wrapProgram $out/bin/pleroma \
+      --set-default RELEASE_TMP "/tmp"
```

Comment:
This surely doesn't persist anything by accident that might be useful?

Asking for `PrivateTmp=yes` mostly ;-)

If so, we should open an upstream issue regardless, to just default to using tempdir mechanisms, and add it in the comment above.

Reply:
Yup, it's safe.

I read the full 3 layers of OTP wrappers to double check :)